src/Reader/Reader.php

```diff
@@ -207,26 +212,27 @@ public static function import($uri, $etag = null, $lastModified = null)
                     $lastModified = $cache->getItem($cacheId . '_lastmodified');
                 }
                 if ($etag) {
-                    $headers->addHeaderLine('If-None-Match', $etag);
+                    $headers['If-None-Match'] = [$etag];
                 }
                 if ($lastModified) {
-                    $headers->addHeaderLine('If-Modified-Since', $lastModified);
+                    $headers['If-Modified-Since'] = [$lastModified];
                 }
             }
-            $response = $client->send();
+            $response = $client->get($uri, $headers);
```

Comment:
> But old clients are not aware of the headers ($headers->addHeaderLine)

First: I have no idea where your reference to `$headers->addHeaderLine` is coming from; there is no headers *object* or *interface* in this patch or in the current source code. There *is* a `HeaderAwareResponseInterface::addHeaderLine` method in this patch, but that method is *only* called following an `instanceof Http\HeaderAwareResponseInterface` check — which is intentional, so that *existing* clients, which could not work with headers, will continue to work.

Adding the `HeaderAware*Interface`s gives the ability for *new* clients to opt-in to the caching functionality, which they previously *could not do*; that's part of the reason for this patch!